### PR TITLE
Add skip_dependencies option to update_dependencies

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- The `update_dependencies` task now supports an `ignore_dependencies` option, which prevents CumulusCI from processing a specific GitHub or managed package dependency (whether direct or transitive). This feature may be useful in installers for packages that extend other packages.

# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/salesforce/UpdateDependencies.py [hunk 1]
```diff
@@ -92,7 +108,9 @@ def _run_task(self):
 
         self.logger.info("Preparing static dependencies map")
         dependencies = self.project_config.get_static_dependencies(
-            self.options["dependencies"], include_beta=self.options["include_beta"]
+            self.options["dependencies"],
+            include_beta=self.options["include_beta"],
+            ignore_deps=self.options.get("skip_dependencies"),
```
[the remark this answers is not in the document]
Comment (davidmreed) [reply cumulusci/tasks/salesforce/UpdateDependencies.py:113]: Very good point. Changed to `ignore` pervasively.

File: docs/tasks.rst [hunk 2]
```diff
@@ -1129,7 +1129,7 @@ Options
 ``-o additional_dependencies ADDITIONALDEPENDENCIES``
 	 *Optional*
 
-	 Include schema from additional GitHub repositories that are not explicit dependencies of this project to build a unified data dictionary. Specify as a list of URLs.
+	 Include schema from additional GitHub repositories that are not explicit dependencies of this project to build a unified data dictionary. Specify as a list of dicts as in project__dependencies in cumulusci.yml. Note: only repository dependencies are supported.
```
[the remark this answers is not in the document]
Comment (davidmreed) [reply docs/tasks.rst]: This is a spurious diff for another task (`generate_data_dictionary`) that _does_ only support repository dependencies.

File: cumulusci/tasks/salesforce/UpdateDependencies.py [hunk 3]
```diff
@@ -92,7 +108,9 @@ def _run_task(self):
...
         )
```
[the remark this answers is not in the document]
Comment (davidmreed) [reply cumulusci/tasks/salesforce/UpdateDependencies.py:112]: Thanks, great callout. Fixed.

File: cumulusci/tasks/salesforce/UpdateDependencies.py [hunk 4]
```diff
@@ -78,6 +83,15 @@ def _init_options(self, kwargs):
                 f"Unsupported value for security_type: {self.options['security_type']}"
             )
 
+        if "ignore_dependencies" in self.options:
+            if any(
+                "github" not in dep and "namespace" not in dep
+                for dep in self.options["ignore_dependencies"]
+            ):
+                raise TaskOptionsError(
+                    "An invalid dependency was specified for ignore_dependencies."
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/UpdateDependencies.py:92]: This isn't a blocker, but this error would be more helpful if we show the input that was invalid and explain why it was invalid.

**Pull request merged**